simpleline/render/in_out_manager.py

```diff
@@ -0,0 +1,241 @@
+# Class for managing input and output for application.
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Author(s): Jiri Konecny <jkonecny@redhat.com>
+#
+
+
+import queue
+import threading
+import sys
+
+from enum import Enum
+
+from simpleline.render import INPUT_PROCESSED, INPUT_DISCARDED
+from simpleline.event_loop import ExitMainLoop
+from simpleline.event_loop.signals import ExceptionSignal, InputReadySignal
+from simpleline.render.prompt import Prompt
+from simpleline.render.widgets import TextWidget
+
+
+RAW_INPUT_LOCK = threading.Lock()
+
+
+class InOutManager(object):
+
+    def __init__(self, event_loop):
+        """Processor for common user input and output.
+
+        :param event_loop: Event loop used for the renderer.
+        :type event_loop: Class based on `simpleline.event_loop.AbstractEventLoop`.
+        """
+        super().__init__()
+        self._input_error_counter = 0
+        self._input_error_threshold = 5
+        self._input_thread = None
+        self._event_loop = event_loop
+        self._getpass_func = None
+        self._width = 80
+
+    @property
+    def input_error_counter(self):
+        """Return counter of user bad inputs."""
+        return self._input_error_counter
+
+    @property
+    def getpass(self):
+        """Return a function for getting passwords."""
+        if not self._getpass_func:
+            import getpass
+            self._getpass_func = getpass.getpass
+        else:
+            return self._getpass_func
```

Comment:
Yes, `pylint` has this problem but only when you are calling function which you are getting from that property on the same line e.g.: ``self.get_pass()``. So this is correct usage for `pylint`.
Good idea for renaming. I'll change that.